File: Data-Mining/assignment3/kmeans/show_digit.py
import numpy as np
import matplotlib.pyplot as plt

def show_digit(x):
    """
    Inputs:
        x: cluster center matrix (k, p), returned by kmeans.
    """

    w = 20
    h = 20
    col = 10
    row = (x.shape[0] + col - 1) // col
    plt.figure(figsize=(10, 10))
    padding = row * col - x.shape[0]
    if padding:
        print(x.shape, padding)
        x = np.vstack((x, np.zeros((padding, x.shape[1]))))
    x = x.reshape(row, col, w, h).transpose(0, 3, 1, 2).reshape(row * h, col * w )
    plt.imshow(x, cmap='gray')

File: Data-Mining/assignment3/kmeans/show_digit.m
function show_digit(X)
%SHOW_IMAGE Show a clustering centers image
%   Inputs:
%       X: cluster center matrix, returned by kmeans.

W = 20;
H = 20;
ShowLine = size(X, 1)/10;
numPerLine = 10;

Y = zeros(H,W);
for i=0:ShowLine-1
for j=0:numPerLine-1
    Y(i*H+1:(i+1)*H,j*W+1:(j+1)*W) = reshape(X(i*numPerLine+j+1,:),[H,W]);
end
end

imagesc(Y);colormap(gray);axis image;
end